packages/squiggle-lang/src/errors/messages.ts

```diff
@@ -132,8 +132,23 @@ export class REDomainError extends ErrorMessage {
     super();
   }
 
+  //It's awkward that we need to check the specific type of error here, given we already needed to calculate this in order to throw the error, but this way we don't need to pass in the error specifics to the constructor.
```

Comment:
Yeah, this is messy.

Suggestion:
1) split `REDomainError` into two, one for types, another for range checks
2) throw an error from `domain.includes`, instead of returning true/false (rename to `checkValue` or something)